pkg/web/oauthclient/cookie_state.go

```diff
@@ -35,13 +35,19 @@ func init() {
 }
 
 // StateCookie returns the cookie storing the state of the console.
-func (oc *OAuthClient) StateCookie() *cookie.Cookie {
+func (oc *OAuthClient) StateCookie(tls bool) *cookie.Cookie {
+	sameSite := http.SameSiteNoneMode
+	if !tls {
+		// SameSite=None will be rejected when not using TLS, so we have to
+		// use SameSite=Lax in this case.
+		sameSite = http.SameSiteLaxMode
+	}
```

Comment:
I really don't like this. See my comments in https://github.com/TheThingsNetwork/lorawan-stack/pull/4286

Reply:
Sorry this bit got into this PR accidentally.